tests/common/include/tests/common/scattering.inl

```diff
@@ -0,0 +1,128 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2023 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Project include(s).
+#include "detray/propagator/actors/pointwise_material_interactor.hpp"
+#include "detray/simulation/detail/scattering_helper.hpp"
+#include "detray/simulation/random_scatterer.hpp"
+#include "detray/tracks/tracks.hpp"
+#include "detray/utils/statistics.hpp"
+
+// google-test include(s).
+#include <gtest/gtest.h>
+
+// System include(s).
+#include <random>
+#include <vector>
+
+using namespace detray;
+using transform3 = __plugin::transform3<detray::scalar>;
+using matrix_operator = typename transform3::matrix_actor;
+using vector3 = typename transform3::vector3;
+
+// Test scattering helper
+TEST(scattering, scattering_helper) {
+
+    std::random_device rd{};
+    std::mt19937_64 generator{rd()};
+    generator.seed(0u);
+
+    const vector3 dir{2.f, 2.f, 4.f};
+    const scalar scattering_angle{0.1f};
+
+    // xaxis of curvilinear plane
+    const vector3 u = unit_vectors<vector3>().make_curvilinear_unit_u(dir);
+
+    // normal vector of the plane defined by u and dir
+    const vector3 w = vector::cross(dir, u);
+
+    std::vector<scalar> angle_diffs;
+
+    std::size_t n_samples{100000u};
+    for (std::size_t i = 0u; i < n_samples; i++) {
+
+        // Scatter the input direction
+        const vector3 new_dir =
+            scattering_helper<transform3>()(dir, scattering_angle, generator);
+
+        // Assign a plus sign if the dot product of new direction and w vector
+        // is plus
+        const scalar sign = vector::dot(w, new_dir) > 0 ? 1.f : -1.f;
+
+        // the cosine of angle between original direction and new one
+        scalar cos_theta = vector::dot(dir, new_dir) /
+                           (getter::norm(dir) * getter::norm(new_dir));
+
+        // To prevent rounding error where cos_theta is out of [-1, 1]
+        cos_theta = std::min(cos_theta, scalar{1.f});
+        cos_theta = std::max(cos_theta, scalar{-1.f});
+
+        // Geht the angle between original direction and new one
+        const scalar angle = sign * std::acos(cos_theta);
+
+        angle_diffs.push_back(angle);
+    }
+
+    EXPECT_NEAR(statistics::mean(angle_diffs), 0.f, 1e-3f);
+
+    // Tolerate upto 1% difference
+    const scalar stddev = std::sqrt(statistics::variance(angle_diffs, 0.f));
```

Comment:
This is calculating the mean deviation from zero, not the stddev, why?

Reply:
It's because we know that `angle_diff` follows the Gaussian distribution whose mean value is zero; As written in `scattering_helper.hpp`, scattering assumes the Gaussian distribution around the original direction

Reply:
In that case ```statistics::variance(angle_diffs)``` should do the trick or what am I missing?

Reply:
The unit test will still pass. But I believe that It is not a correct way to calculate the varaince.
`statistics::variance(angle_diffs)` will use the mean value of samples which is close to zero but not zero.

What I claim is that we need to use the known mean value rather than the sampled one.

Reply:
Thanks for checking